operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/primarytosecondaydependent/PrimaryToSecondaryDependentReconciler.java

```diff
@@ -0,0 +1,85 @@
+import java.util.stream.Collectors;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.javaoperatorsdk.operator.api.config.informer.InformerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.*;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.Dependent;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.PrimaryToSecondaryMapper;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+import io.javaoperatorsdk.operator.support.TestExecutionInfoProvider;
+
+import static io.javaoperatorsdk.operator.sample.primarytosecondaydependent.PrimaryToSecondaryDependentReconciler.CONFIG_MAP;
+import static io.javaoperatorsdk.operator.sample.primarytosecondaydependent.PrimaryToSecondaryDependentReconciler.CONFIG_MAP_EVENT_SOURCE;
+
+/**
+ * Sample showcases how it is possible to do a primary to secondary mapper for a dependent resource.
+ * Note that this is usually just used with read only resources. So it has limited usage, one reason
+ * to use it is to have nice condition on that resource within a workflow.
+ */
+@ControllerConfiguration(dependents = {@Dependent(type = ConfigMapDependent.class,
+    name = CONFIG_MAP,
+    reconcilePrecondition = ConfigMapReconcilePrecondition.class,
+    useEventSourceWithName = CONFIG_MAP_EVENT_SOURCE),
+    @Dependent(type = SecretDependent.class, dependsOn = CONFIG_MAP)})
+public class PrimaryToSecondaryDependentReconciler
+    implements Reconciler<PrimaryToSecondaryDependentCustomResource>, TestExecutionInfoProvider,
+    EventSourceInitializer<PrimaryToSecondaryDependentCustomResource> {
+
+  public static final String DATA_KEY = "data";
+  public static final String CONFIG_MAP = "ConfigMap";
+  public static final String CONFIG_MAP_INDEX = "ConfigMapIndex";
+  public static final String CONFIG_MAP_EVENT_SOURCE = "ConfigMapEventSource";
+
+  private final AtomicInteger numberOfExecutions = new AtomicInteger(0);
+
+  @Override
+  public UpdateControl<PrimaryToSecondaryDependentCustomResource> reconcile(
+      PrimaryToSecondaryDependentCustomResource resource,
+      Context<PrimaryToSecondaryDependentCustomResource> context) {
+    numberOfExecutions.addAndGet(1);
+    return UpdateControl.noUpdate();
+  }
+
+  public int getNumberOfExecutions() {
+    return numberOfExecutions.get();
+  }
+
+  /**
+   * Creating an Event Source and setting it for the Dependent Resource. Since it is not possible to
+   * do this setup within the bounds of the KubernetesDependentResource API. However, this is quite
+   * a corner case; might be covered more out of the box in the future if there will be demand for
+   * it.
+   **/
+  @Override
+  public Map<String, EventSource> prepareEventSources(
+      EventSourceContext<PrimaryToSecondaryDependentCustomResource> context) {
```

Comment:
It would be helpful to add comments here as to how the different resources depend on one another and how the mappers make these relations explicit.

Reply:
not sure what do you mean, you mean how the secondary depends on primary? and why the mappers needed?

Reply:
Explain how the secret is found from the CR via the mapper and why the mapper is needed, yes, as this would help users figure out how this scenario can (or not) be applied to their own use case.

Reply:
I guess you mean the config map. kk.

Reply:
added